### DISCUSSION TITLE
Is there a way to trigger a sort programmatically?

### DISCUSSION THREAD

**Discussion by m1212e** [cuire/svelte-grid-extended#59] [Q&A]
In a scenario I currently have, I want to place a few clickable items in the grid and when clicked, they should grow, revealing some content. I want the other elements to intelligently flow around the expanded item. But when adjusting the size of a single element it just overflows into the others, which don't move. But when I expand them programmatically, it doesn't register a user interaction event and therefore does not rearrange the grid items. Also, it would be very cool to have an option to arrange the items in a way they take the least amount of space.

Is something like this possible?

**Comment by cuire** [comment]
Hey. Now there is no work around to achieve that. This problem is related to #57, I need to move sorting logic from `GridItem` to `Grid` responsibility. 
The last 3 weeks at work have been terrible and I haven't had time to work on the project. Ill try to fix thit in a week and probably Ill open a new issue.

**Reply by m1212e** [reply]
No worries, it's a free library, do as much as you would like to. Thanks for your work!
